hubble/executor/hubio.py

```diff
@@ -1271,14 +1295,16 @@ def pull(self) -> str:
 
                 build_env = executor.build_env
 
-                presented_id = executor.name if executor.name else executor.uuid
-                executor_name = (
-                    f'{presented_id}'
-                    if executor.visibility == 'public' or not secret
-                    else f'{presented_id}:{secret}'
-                ) + (f'/{tag}' if tag else '')
+                if executor.visibility == 'public' or not secret:
+                    executor_name = name
+                else:
+                    executor_name = f'{name}:{secret}'
+
+                is_legacy_uri = scheme.startswith('jinahub')
+                if tag:
+                    executor_name += f'/{tag}' if is_legacy_uri else f':{tag}'
 
-                if scheme == 'jinahub+docker':
+                if scheme.endswith('+docker'):
```

Comment:
This is buggy?

Reply:
Buggy how? Could you elaborate more, please?